scala-api/src/main/scala/com/cloudera/livy/LivyScalaClient.scala

```diff
@@ -0,0 +1,65 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.cloudera.livy
+
+import java.io.File
+import java.net.URI
+import java.util.concurrent.{ScheduledThreadPoolExecutor}
+
+import scala.concurrent._
+
+class LivyScalaClient(livyJavaClient: LivyClient, threadPoolSize: Integer) {
+
+  private val executor = new ScheduledThreadPoolExecutor(threadPoolSize)
+
+  def this(livyJavaClient: LivyClient) {
+    this(livyJavaClient, 2)
+  }
+
+  def submit[T](block: ScalaJobContext => T): ScalaJobHandle[T] = {
+    val job = new Job[T] {
+      @throws(classOf[Exception])
+      override def call(jobContext: JobContext): T = block(new ScalaJobContext(jobContext))
```

Comment:
Nevermind it's the function's argument. I'd call it `fn` or `job` which are more common names for these arguments.